mapsforge-map/src/main/java/org/mapsforge/map/datastore/MultiMapDataStore.java

```diff
@@ -44,13 +48,13 @@ public enum DataPolicy {
 
     private BoundingBox boundingBox;
     private final DataPolicy dataPolicy;
-    private final List<MapDataStore> mapDatabases;
+    private final Set<MapDataStore> mapDatabases;
     private LatLong startPosition;
     private byte startZoomLevel;
 
     public MultiMapDataStore(DataPolicy dataPolicy) {
         this.dataPolicy = dataPolicy;
-        this.mapDatabases = new ArrayList<>();
+        this.mapDatabases = new TreeSet<>();
```

Comment:
@Sublimis Why this change?

If the order of insertion is still important, we can sort the list in the app or in the library.

Reply:
Done that. Collections.sort() is guaranteed to be stable, so it will not change the insertion order if all map files happen to have the same priority (which is the default, so it retains the current behavior).